internal/exec/packer/packer.go

```diff
@@ -165,7 +166,33 @@ func (b *Builder) makeNonNullPacker(schemaType types.Type, reflectType reflect.T
 	}
 }
 
+type packable struct {
+	Name    string
+	Type    types.Type
+	Default types.Value
+}
+
 func (b *Builder) MakeStructPacker(values []*types.InputValueDefinition, typ reflect.Type) (*StructPacker, error) {
+	p := make([]packable, len(values))
+	for i, v := range values {
+		p[i] = packable{Name: v.Name.Name, Type: v.Type, Default: v.Default}
+	}
+
+	return b.makeStructPacker(p, typ)
+}
+
+func (b *Builder) MakeStructPackerArgs(values []*types.Argument, typ reflect.Type) (*StructPacker, error) {
+	p := make([]packable, len(values))
+	for i, v := range values {
+		// TODO: does a default apply here?
+		// FIXME: What should the type actually be?
+		p[i] = packable{Name: v.Name.Name, Type: &types.ScalarTypeDefinition{}}
```

Comment:
Driective args are of type `types.ArgumentsDefinition` in my opinion. https://spec.graphql.org/draft/#sec-Type-System.Directives